File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/ProverSearchCredentialsTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class ProverSearchCredentialsTest : AnonCredsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestProverSearchCredentialsWorksForEmptyFilter()
        {           
            var credentials = await AnonCreds.ProverSearchCredentialsAsync(wallet, "{}");
            Assert.AreEqual(3, credentials.TotalCount);

            var credentialsArray = await credentials.NextAsync(100);
            var jsonArray = JArray.Parse(credentialsArray);

            Assert.AreEqual(3, jsonArray.Count);

            //TODO: Shouldn't there be an explicit close of the credential search here?
        }

        [TestMethod]
        public async Task TestProverSearchCredentialsWorksForInvalidFilterJson()
        {
            var filter = "{\"issuer_id\":1}";
            var ex = await Assert.ThrowsExceptionAsync<WalletInvalidQueryException>(() =>
                AnonCreds.ProverSearchCredentialsAsync(wallet, filter)
            );
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/ProverSearchCredentialsTest.java
package org.hyperledger.indy.sdk.anoncreds;


import org.json.JSONArray;
import org.json.JSONObject;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ProverSearchCredentialsTest extends AnoncredsIntegrationTest {

	@Test
	public void testProverSearchCredentialsWorksForEmptyFilter() throws Exception {

		JSONObject json = new JSONObject();
		String filter = json.toString();

		CredentialsSearch credentials = CredentialsSearch.open(wallet, filter).get();
		assertEquals(3, credentials.totalCount());

		JSONArray credentialsArray = new JSONArray(credentials.fetchNextCredentials(100).get());

		assertEquals(3, credentialsArray.length());
	}
}
